### PR TITLE
Reset NFT state before bridging (with _burn)

### PR DESCRIPTION
**Pull request by vminkov**
Reset NFT state before bridging (with _burn)

the changes should make it so that before bridging, the burn calls that can only be made by the bridge, reset or update the storage data that is related to the _tokenId that is burnt

### CODE REVIEW

File: contracts/VoteEscrow.sol [hunk 1]
```diff
@@ -484,20 +491,38 @@ contract VoteEscrow is XERC721Upgradeable, IVotesUpgradeable, ReentrancyGuardUpg
...
+  function _afterMint(uint256 _tokenId, bytes memory _metadata) internal virtual override {
+    (int128 amount, uint256 end) = abi.decode(_metadata, (int128, uint256));
+    LockedBalance memory newLocked = LockedBalance(amount, end);
+    locked[_tokenId] = newLocked;
+    LockedBalance memory oldLocked = LockedBalance(0, 0);
+    _checkpoint(_tokenId, oldLocked, newLocked);
+  }
+
+  function _beforeBurn(uint256 _tokenId) internal virtual override returns (bytes memory _metadata) {
+    _metadata = abi.encode(locked[_tokenId].amount, locked[_tokenId].end);
```
Comment (qzirak) [inline contracts/VoteEscrow.sol:510]: What are we  doing with this '_metadata'  ?

Comment (vminkov) [reply contracts/VoteEscrow.sol:510]: the bridge is going to use it in `XERC721Upgradeable` to mint the same NFT on the other chain (using the metadata)

Comment (qzirak) [reply contracts/VoteEscrow.sol:510]: I meant are we emitting the _metadata or something ? How are we handling the return value of the function ?

Comment (vminkov) [reply contracts/VoteEscrow.sol:510]: yes, we are handling it in `XERC721Upgradeable::burn()`

Comment (vminkov) [reply contracts/VoteEscrow.sol:510]: oh, sorry, these changes are not there

File: contracts/XERC721Upgradeable.sol [hunk 2]
```diff
@@ -49,12 +51,15 @@ abstract contract XERC721Upgradeable is ERC721Upgradeable, Ownable2StepUpgradeab
...
+
+    emit MintAsBridge(_tokenId, _metadata);
   }
 
-  // specify the locked assets amount and unlock time metadata
   function burn(uint256 _tokenId) public onlyBridge returns (bytes memory _metadata) {
     _metadata = _beforeBurn(_tokenId);
     _burn(_tokenId);
+
+    emit BurnAsBridge(_tokenId, _metadata);
```
Comment (vminkov) [inline contracts/XERC721Upgradeable.sol:62]: @qzirak here is the metadata being emitted

File: contracts/VoteEscrow.sol (hunk 1 above)
Comment (vminkov) [reply contracts/VoteEscrow.sol:508]: https://github.com/vminkov/veXYZ/pull/17/files#r1300143813

File: contracts/XERC721Upgradeable.sol (hunk 2 above)
Comment (vminkov) [inline contracts/XERC721Upgradeable.sol:62]: here is the metadata event @qzirak

**Pull request merged**